src/contributors/MetricsTableContributor.ts

```diff
@@ -152,8 +159,81 @@ export class MetricsTableContributor extends Contributor {
     /** @override */
     /** todo !!!! specify data type and return type  */
     public computeData(data: Array<MetricsTableResponse>): MetricsTable {
-        console.log('test is ok');
-        return null;
+        // todo: to be improved
+        const headers = new Map();
+        const rows: Map<string, MetricsTableRow> = new Map();
+        const maxCount = new Map();
+        let rowDataMaxLength = 0;
+
+        data = this.orderMetricsTableResponse(data);
+        data.forEach(metricsResponse => {
+            metricsResponse.aggregationResponse.elements.forEach(elements => {
+                elements.metrics.forEach(metrics => {
+                    const uniqColumn = `${metricsResponse.collection}_${metrics.field}_${metrics.type}`;
+                    const uniqKeyForSum = `${elements.key_as_string}_${metrics.type}`;
+                    // storing headers;
+                    if(!headers.has(uniqColumn)) {
+                        const headerItem: MetricsTableHeader = {
+                            title: metricsResponse.collection ,
+                            subTitle: metrics.field,
+                            metric: metrics.type
+                        };
+                        headers.set(uniqColumn, headerItem);
+                    }
+
+                    // storing uniq row;
+                    if(rows.has(uniqKeyForSum)) {
+                        rows.get(uniqKeyForSum).data.push({maxValue: 0, value: metrics.value});
+                    } else {
+                        const metricsTableData: MetricsTableData = {maxValue: 0, value: metrics.value};
+                        const metricsTableRow: MetricsTableRow = {data: [], term: elements.key_as_string};
+                        metricsTableRow.data.push(metricsTableData);
+                        rows.set(uniqKeyForSum, metricsTableRow);
+                    }
+
+                    if(rowDataMaxLength <  rows.get(uniqKeyForSum).data.length){
+                        rowDataMaxLength = rows.get(uniqKeyForSum).data.length;
+                    }
+
+                    // storing uniq sum for each value;
+                    // externalise this method
+                    // in case of not count
+                    if(maxCount.has(uniqKeyForSum) &&  maxCount.get(uniqKeyForSum) <  metrics.value) {
+                       maxCount.set(uniqKeyForSum, metrics.value);
+                    } else {
+                        maxCount.set(uniqKeyForSum,  metrics.value);
+                    }
+                });
+            });
+        });
+
+        const metricsTable: MetricsTable = {data: [], header: []};
+        // att the end we setHeaders
+        for (const value of headers.values()){
+            metricsTable.header.push(value);
+        }
+
+        // att the end we update rows
+        for (const  [key, metricTableRow] of rows.entries()){
+            // update the max count
+            metricTableRow.data.forEach(data => {
```

Comment:
I think that the map won't respect the order of keys as driven by the first column :/ because maps use alphabetical order